src/fal/dag.py

```diff
@@ -0,0 +1,129 @@
+from dataclasses import dataclass, field
```

Comment:
Maybe we could write some tests for this file?

Reply:
Not specific ones, just passing in examples and checking that the results make sense.

Reply:
yea we should